### DISCUSSION TITLE
[V3] Override for isOnline and isDocumentVisible

### DISCUSSION THREAD

**Discussion by xorob0** [TanStack/query#1230] [Ideas]
Hi everyone 👋

We are setting up a project in React Native that would need to work offline. We saw that the V3 will have offline support, but the implementation of [`isOnline`](https://github.com/tannerlinsley/react-query/blob/e200c99496b7ebea84da5a0df38242d80554a844/src/core/utils.ts#L111-L113) and [`isDocumentVisible`](https://github.com/tannerlinsley/react-query/blob/e200c99496b7ebea84da5a0df38242d80554a844/src/core/utils.ts#L103-L109) depend on the browser APIs and would not work on React Native. It would be nice if we could simply override those implementation with our owns when initializing the environment.

I see there is already something similar for the[ `console`](https://github.com/tannerlinsley/react-query/blob/e200c99496b7ebea84da5a0df38242d80554a844/src/core/utils.ts#L43-L47)

**Comment by boschni** [comment]
Hi @xorob0! This should be possible in V3 with the `onlineManager` and `focusManager` instances: https://react-query-beta.tanstack.com/reference/onlineManager

**Reply by xmflsct** [reply]
Just tried `onlineManager.setOnline(false)` manually, but it doesn't affect any of the existing behaviours? Could not find more information in the documentation therefore asked here.

**Reply by boschni** [reply]
Hi @xmflsct! It should not retry while online is set to false. Do you expect it to do something different?

**Reply by xmflsct** [reply]
I played around a bit more, I think it is the most reasonable solution as of now. 👍
